frontend/src/metabase/visualizations/visualizations/CartesianChart/use-chart-events.ts

```diff
@@ -322,16 +323,12 @@ export const useChartEvents = (
         settings,
       };
 
-      if (
-        !areMultipleCards &&
-        hasBreakout &&
-        visualizationIsClickable(clickData)
-      ) {
+      if (hasBreakout && visualizationIsClickable(clickData)) {
         onVisualizationClick({
           ...clickData,
           element: event.currentTarget,
         });
-      } else {
+      } else if (isDashboard) {
```

Comment:
We have to add this `isDashboard` check to the else conditional, or else that function will be fired in the query builder, which is the thing that is causing the invalid request, since `seriesModel.cardId` is undefined in the query builder